### PR TITLE
pkg/api: Move service helpers to pkg/apis/core

### PR DESCRIPTION
**Pull request by sttts**
Follow-up of https://github.com/kubernetes/kubernetes/pull/53651. Part of https://github.com/kubernetes/kubernetes/issues/58292.

### CODE REVIEW

File: pkg/kubectl/.import-restrictions [hunk 1]
```diff
@@ -8,9 +8,9 @@
                 "k8s.io/kubernetes/pkg/api/pod",
                 "k8s.io/kubernetes/pkg/api/ref",
                 "k8s.io/kubernetes/pkg/api/resource",
-                "k8s.io/kubernetes/pkg/api/service",
+                "k8s.io/kubernetes/pkg/apis/core/helper/service",
```
Comment (deads2k) [inline pkg/kubectl/.import-restrictions]: Why?  Just tidiness?  The registry one made a  lot of sense, but this appears more spread out.  Are there more coming?

Comment (sttts) [reply pkg/kubectl/.import-restrictions]: It's a core api helper and should live in some core package. Midterm goal is to clean up pkg/api from any legacy api code, possibly declare pkg/api as a integration package only and maybe some day get rid of it.

Comment (deads2k) [reply pkg/kubectl/.import-restrictions]: > It's a core api helper and should live in some core package. Midterm goal is to clean up pkg/api from any legacy api code, possibly declare pkg/api as a integration package only and maybe some day get rid of it.

Ok.  If we're going to the effort of moving them, why not `pkg/apihelpers/<group>` so changes don't require api reviewers approval (these aren't changing types) and we can see just how many we have.

**Pull request closed without merging**